lib/open-uri.rb

```diff
@@ -27,7 +27,18 @@ class << self
   # We can accept URIs and strings that begin with http://, https:// and
   # ftp://. In these cases, the opened file object is extended by OpenURI::Meta.
   def open(name, *rest, &block) # :doc:
-    if name.respond_to?(:open)
+    if name.respond_to?(:to_io)
```

Comment:
With this change, `open` goes here for not only `Tempfile` but also `Socket`.

```
irb(main):023:0> Socket.tcp("www.ruby-lang.org", 80) {|sock| p sock.respond_to?(:to_io); }
true
```

I think it is not your intended behavior.